platform/darwin/src/MGLFeature_Private.h

```diff
@@ -46,6 +47,10 @@ NS_ASSUME_NONNULL_END
             NSSet<Class> *identifierClasses = [NSSet setWithArray:@[[NSString class], [NSNumber class]]]; \
             identifier = [decoder decodeObjectOfClasses:identifierClasses forKey:@"identifier"]; \
             attributes = [decoder decodeObjectOfClass:[NSDictionary class] forKey:@"attributes"]; \
+            /* If this feature is a cluster, then change the class */ \
+            if (MGLFeatureHasClusterAttribute(self)) { \
+                MGLConvertFeatureToClusterSubclass(self); \
```

Comment:
It feels pretty hacky to change the class within an initializer after calling the superclass’s initializer. In principle, this should only be necessary if we expect that a non-MGLCluster could be archived then unarchived as an MGLCluster, which seems unlikely.